pkg/plugins/resources/dockerimage/main.go

```diff
@@ -70,14 +72,64 @@ func New(spec interface{}) (*DockerImage, error) {
 
 	newResource.options = append(newResource.options, remote.WithAuthFromKeychain(authn.NewMultiKeychain(keychains...)))
 
-	if len(newSpec.Architectures) == 1 {
-		newResource.options = append(newResource.options, remote.WithPlatform(v1.Platform{Architecture: newSpec.Architectures[0], OS: "linux"}))
-	}
-
 	return newResource, nil
 }
 
 // Changelog returns the changelog for this resource, or an empty string if not supported
 func (di *DockerImage) Changelog() string {
 	return ""
 }
+
+func (di *DockerImage) createRef(source string) (name.Reference, error) {
+	refName := di.spec.Image
+	switch di.spec.Tag == "" {
+	case true:
+		refName += ":" + source
+	case false:
+		refName += ":" + di.spec.Tag
+	}
+
+	ref, err := name.ParseReference(refName)
+	if err != nil {
+		return nil, fmt.Errorf("invalid image %s: %w", refName, err)
+	}
+
+	return ref, nil
+}
+
+// checkImage checks if a container reference exists on the "remote" registry with a given set of options
+func (di *DockerImage) checkImage(ref name.Reference, arch string) (bool, error) {
+	remoteOptions := append(di.options, remote.WithPlatform(v1.Platform{Architecture: arch, OS: "linux"}))
+	descriptor, err := remote.Get(ref, remoteOptions...)
+	if err != nil {
+		if strings.Contains(err.Error(), "unexpected status code 404") {
+			logrus.Infof("%s The Docker image %s doesn't exist.",
+				result.FAILURE,
+				ref.Name(),
+			)
+			return false, nil
+		}
+		return false, err
+	}
+
+	_, err = descriptor.Image()
```

Comment:
This was the important part the `WithPlatform` were not being checked until this call.